File: synapse-transport/src/main/java/com/synapse/transport/feature/experiment/dto/response/ExperimentInferenceResponse.java
package com.synapse.transport.feature.experiment.dto.response;

import java.util.UUID;

import com.synapse.transport.common.enums.InferenceStatus;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import tools.jackson.databind.JsonNode;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ExperimentInferenceResponse {
    private UUID experimentId;
    private InferenceStatus status;

    private Integer sampleNumber;
    private JsonNode modelIr;
    private JsonNode inferenceResult;

    private String lastInferenceError;
}

File: synapse-ui/src/types/api/experiment/experiment-inference.response.ts
import type { InferenceStatus } from "../../enums/transport.enums";
import type { InferenceResult } from "../../runtime/inference/inference-result.type";
import type { ModelIR } from "../../runtime/model/model-ir.type";

export interface ExperimentInferenceResponse {
  experimentId: string;
  status: InferenceStatus;
  sampleNumber: number | null;
  modelIr: ModelIR | null;
  inferenceResult: InferenceResult | null;
  lastInferenceError: string | null;
}
